tools/Dashboard/DevHome.Dashboard/Controls/WidgetControl.xaml.cs

```diff
@@ -18,21 +18,28 @@ public sealed partial class WidgetControl : UserControl
 {
     private MenuFlyoutItem _currentSelectedSize;
 
-    public WidgetControl()
-    {
-        this.InitializeComponent();
-        ActualThemeChanged += OnActualThemeChanged;
-    }
-
     public WidgetViewModel WidgetSource
     {
         get => (WidgetViewModel)GetValue(WidgetSourceProperty);
-        set => SetValue(WidgetSourceProperty, value);
+        set
+        {
+            SetValue(WidgetSourceProperty, value);
+            if (WidgetSource != null)
+            {
+                UpdateWidgetHeaderIconFillAsync();

```

Comment:
NIT: Add a comment mentioning the side-effect of calling UpdateWidgetHeaferIconFillAsync.

Reply:
Spoke offline.  The intention of my comment was misunderstood because it was on linen 29.  I was referring to the set method as a whole, and not this specific method.